src/ray/core_worker/core_worker.cc

```diff
@@ -93,6 +101,34 @@ CoreWorker::CoreWorker(
   }
 }
 
+void CoreWorker::AddActiveObjectID(const ObjectID &object_id) {
+  io_service_.post([this, object_id]() -> void { active_object_ids_.insert(object_id); });
+}
+
+void CoreWorker::RemoveActiveObjectID(const ObjectID &object_id) {
+  io_service_.post([this, object_id]() -> void {
+    if (!active_object_ids_.erase(object_id)) {
+      RAY_LOG(WARNING) << "Tried to erase non-existent object ID" << object_id;
+    }
+  });
+}
+
+void CoreWorker::SendActiveObjectIDsHeartbeat() {
+  RAY_LOG(DEBUG) << "Sending " << active_object_ids_.size() << " object IDs to raylet.";
+  if (active_object_ids_.size() > RayConfig::instance().raylet_active_object_ids_size()) {
+    RAY_LOG(WARNING) << active_object_ids_.size() << "object IDs are currently in scope. "
```

Comment:
Won't this be too noisy?

Reply:
This shouldn't ever happen under normal operation - if it does, we probably want a bug report so it's fine to be noisy IMO.